railties/lib/rails/test_unit/minitest_plugin.rb

```diff
@@ -59,19 +59,18 @@ def self.plugin_rails_options(opts, options)
 
     options[:color] = true
     options[:output_inline] = true
-    options[:patterns] = defined?(@rake_patterns) ? @rake_patterns : opts.order!
+    options[:patterns] = opts.order! unless run_via[:rake]
   end
 
-  # Running several Rake tasks in a single command would trip up the runner,
-  # as the patterns would also contain the other Rake tasks.
   def self.rake_run(patterns) # :nodoc:
-    @rake_patterns = patterns
+    ::Rails::TestRequirer.require_files patterns
+    run_via[:rake] = true
```

Comment:
Think it makes more sense to assign `run_via` as the first step and then require the files.